File: Python/Esercitazione10/materiale_pericoloso.py
from materiale import Materiale

class MaterialePericoloso(Materiale):
	def __init__(self, nome: str, quantita: int, prezzo_unitario: float, classe_pericolo : int):
		super().__init__(nome, quantita, prezzo_unitario)
		self._classe_pericolo = classe_pericolo

	def __repr__(self):
		return f"Materiale[nome='{self._nome}'; quantità={self._quantita}; prezzo unitario={self._prezzo_unitario}; classe pericolo={self._classe_pericolo}]"

File: Java/Esercitazione10/MaterialePericoloso.java
package Esercitazione10;

public class MaterialePericoloso extends Materiale {
    private final int classePericolo;


    public MaterialePericoloso(String nome, int quantita, double prezzoUnitario, int classePericolo) {
        super(nome, quantita, prezzoUnitario);
        this.classePericolo = classePericolo;
    }


    @Override
    public String toString() {
        return "Materiale{" +
                "nome='" + super.getNome() + '\'' +
                ", quantita=" + super.getQuantita() +
                ", prezzoUnitario=" + super.getPrezzoUnitario() +
                ", classePericolo=" + this.classePericolo +
        '}';
    }
}
